storm-core/src/jvm/backtype/storm/windowing/EventImpl.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package backtype.storm.windowing;
+
+class EventImpl<T> implements Event<T> {
```

Comment:
Is there a need to have the class EventImpl and the interface Event? Is it reasonable to think that there will ever be another implementation of Event. If not, which in my opinion is the case, we should have only the class Event which provides the implementation. No need for the interface. In the article pasted bellow it says that a good  for when there should be no need for an interface and an implementing class is when one cannot come up with a better name for the class rather than adding the suffix Impl.

http://stackoverflow.com/questions/2814805/java-interfaces-implementation-naming-convention